Formula/aws-sdk-cpp.rb

```diff
@@ -0,0 +1,90 @@
+require "set"
+
+class AwsSdkCpp < Formula
+  desc "AWS SDK for C++"
+  homepage "https://github.com/aws/aws-sdk-cpp"
+  url "https://github.com/aws/aws-sdk-cpp/archive/0.10.1.tar.gz"
+  sha256 "e571382dbccca0c37f38d097a12bf54874190a312e55ce22534cc6b1ddb8ffaf"
+  head "https://github.com/aws/aws-sdk-cpp.git"
+
+  WITHOUT_OPTIONS = {
+    "compute" => ["ec2", "elasticbeanstalk", "lambda"],
+    "storage" => ["s3", "cloudfront", "elasticfilesystem", "glacier", "importexport", "storagegateway"],
+    "database" => ["rds", "dynamodb", "elasticache", "redshift", "dms", "sdb"],
+    "networking" => ["directconnect", "route53", "route53domains"],
+    "devtools" => ["codecommit", "codedeploy", "codepipeline"],
+    "management" => ["cloudformation", "cloudtrail", "config", "opsworks", "ecr", "ecs", "autoscaling", "support", "logs", "monitoring", "ssm"],
+    "security" => ["identity-management", "access-management", "inspector", "waf", "iam", "cloudhsm", "acm", "ds", "kms", "sts"],
+    "analytics" => ["datapipeline", "kinesis", "firehose", "marketplacecommerceanalytics"],
+    "mobile" => ["cognito-identity", "cognito-sync", "devicefarm", "mobileanalytics", "sns"],
+    "appservices" => ["apigateway", "cloudsearch", "cloudsearchdomain", "elastictranscoder", "sqs", "swf", "email"],
+    "other" => ["elasticloadbalancing", "elasticmapreduce", "es", "events", "machinelearning", "queues", "transfer", "iot", "gamelift", "workspaces", "meteringmarketplace"],
+  }.freeze
+
+  WITHOUT_OPTIONS.each { |category, clients| option "without-#{category}", "Exclude clients: #{clients.join(", ")}" }
```

Comment:
@mikemcquaid Can you help me understand better how the bottle/cellar stuff works? In our use case, pulling the already built binaries would be great, but we need to get the static libraries. From what I can tell, the SDK build script does not build shared libraries when the static option is turned on. Can we store a bottle for both the `with-static` and regular shared builds?

Removing the flags and not making the static libraries available in the bottle (if I understand the notion of bottle correctly) would render this unusable for our project (https://github.com/facebook/osquery) as we couldn't have a half hour build of the whole SDK in our dependency init script.